commands/bmc.nicole

```diff
@@ -152,14 +152,45 @@ function cmd_datetime_method {
 # Set system time manually
 #   YYYY-MM-DD-HH-mm-SS - Date and time to set
 function cmd_datetime_set {
+  local datestr="${1-}"
+
   # Only check format here, values are checked for sanity by `date`
-  if ! [[ ${1:-} =~ ^[0-9]{4}(-[0-9]{2}){5}$ ]]; then
+  if ! [[ "${datestr}" =~ ^[0-9]{4}(-[0-9]{2}){5}$ ]]; then
     echo "Please enter date and time in YYYY-MM-DD-HH-mm-SS format" >&2
     return 1
   fi
 
-  local new_date="${1:0:10}"
-  local new_time="${1:11}"
+  local YY
+  local MM
+  local DD
+  local HH
+  local mm
+  local SS
+  IFS='-' read YY MM DD HH mm SS < <(echo ${datestr})
+
+  # Problem Y2k38 isn't resolved in OpenBMC yet
+  if [[ $YY -le 2021 ]] || [[ $YY -gt 2037 ]]; then
```

Comment:
Probably `-lt`. Or `2021` is also prohibited? ;)

Reply:
Fixed